platform/lf_nrf52_support.c

```diff
@@ -143,19 +177,11 @@ int lf_clock_gettime(instant_t* t) {
         // errno = EFAULT; //TODO: why does this not work with new build process?
         return -1;
     }
-    
-    NRF_TIMER3->TASKS_CAPTURE[1] = 1;
-
-    // Handle possible overflow.
-    uint32_t current_timer_time = NRF_TIMER3->CC[1];
-    if (current_timer_time < _lf_previous_timer_time) {
-        // Overflow has occurred. Use _lf_time_epoch_offset to correct.
-        _lf_time_epoch_offset += (1LL << 32) * 1000;
-    }
-
-    *t = ((instant_t)current_timer_time) * 1000 + _lf_time_epoch_offset;
-
-    _lf_previous_timer_time = current_timer_time;
+    uint32_t curr_time;
+    // capture latest timer value
+    curr_time = nrfx_timer_capture(&g_lf_timer_inst, NRF_TIMER_CC_CHANNEL1);
+    *t = ((instant_t)curr_time) * 1000 + _lf_time_epoch_offset;
+    _lf_previous_timer_time = curr_time;
```

Comment:
I think this code still needs to correct the returned value in *t if current_timer_time < _lf_previous_timer_time because there is a race condition. Suppose the timer hits `~0x0` and triggers an interrupt but before the interrupt is handled, some piece of code calls `lf_clock_gettime`.  Then the `_lf_time_epoch_offset` will not have been updated and the time value returned will be off by `(1LL << 32)` microseconds. This could cause the LF program to freeze.  This would probably be a very rare event, so it's not likely to show up in testing. Careful commenting of the code is important.

If I am wrong about this, then `_lf_previous_timer_time` is no longer used in this code and should be removed.

Reply:
I've added the check back and a flag to make sure the interrupt does add to the epoch_offset twice. I might be missing something however, I'll think about this issue more.